packages/material-ui/src/Tabs/Tabs.js

```diff
@@ -239,175 +245,147 @@ class Tabs extends React.Component {
     if (tabMeta.left < tabsMeta.left) {
       // left side of button is out of view
       const nextScrollLeft = tabsMeta.scrollLeft + (tabMeta.left - tabsMeta.left);
-      this.scroll(nextScrollLeft);
+      scroll(nextScrollLeft);
     } else if (tabMeta.right > tabsMeta.right) {
       // right side of button is out of view
       const nextScrollLeft = tabsMeta.scrollLeft + (tabMeta.right - tabsMeta.right);
-      this.scroll(nextScrollLeft);
+      scroll(nextScrollLeft);
     }
-  };
-
-  scroll = value => {
-    animate('scrollLeft', this.tabsRef, value);
-  };
-
-  updateScrollButtonState = () => {
-    const { scrollButtons, theme, variant } = this.props;
-    const scrollable = variant === 'scrollable';
+  });
 
+  const updateScrollButtonState = useEventCallback(() => {
     if (scrollable && scrollButtons !== 'off') {
-      const { scrollWidth, clientWidth } = this.tabsRef;
-      const scrollLeft = getNormalizedScrollLeft(this.tabsRef, theme.direction);
+      const { scrollWidth, clientWidth } = tabsRef.current;
+      const scrollLeft = getNormalizedScrollLeft(tabsRef.current, theme.direction);
 
       // use 1 for the potential rounding error with browser zooms.
-      const showLeftScroll =
-        theme.direction === 'rtl' ? scrollLeft < scrollWidth - clientWidth - 1 : scrollLeft > 1;
-      const showRightScroll =
-        theme.direction !== 'rtl' ? scrollLeft < scrollWidth - clientWidth - 1 : scrollLeft > 1;
-
-      if (
-        showLeftScroll !== this.state.showLeftScroll ||
-        showRightScroll !== this.state.showRightScroll
-      ) {
-        this.setState({ showLeftScroll, showRightScroll });
+      const showLeftScroll = isRtl ? scrollLeft < scrollWidth - clientWidth - 1 : scrollLeft > 1;
+      const showRightScroll = !isRtl ? scrollLeft < scrollWidth - clientWidth - 1 : scrollLeft > 1;
+
+      if (showLeftScroll !== displayScroll.left || showRightScroll !== displayScroll.right) {
+        setDisplayScroll({ left: showLeftScroll, right: showRightScroll });
       }
     }
-  };
+  });
 
-  updateIndicatorState(props) {
-    const { theme, value } = props;
+  React.useEffect(() => {
+    const handleResize = debounce(() => {
+      updateIndicatorState();
+      updateScrollButtonState();
+    });
 
-    const { tabsMeta, tabMeta } = this.getTabsMeta(value, theme.direction);
-    let left = 0;
+    window.addEventListener('resize', handleResize);
```

Comment:
Probably doesn't matter because this is exactly equivalent to the old implementation but in the past we had issues in multi-window (`window.open()`) environments where `window` was referring to the parent window, and so the most bulletproof way was to do `element.ownerDocument.defaultView` to get the right `window` object (i.e. child not parent). I'm aware of the `ownerWindow` helper https://github.com/mui-org/material-ui/blob/89687f38cae750650555772ba4d821c9084d8dfc/packages/material-ui/src/utils/ownerWindow.js#L5 btw. It's just when I last looked at the implementation, I wasn't too sure why it falls back the way it does - `parentView` isn't a DOM property from what I can tell, nor is [`parentWindow`](https://github.com/react-bootstrap/dom-helpers/blob/4cb2993f5ac074a051ad8e951b34c2e47e2644d1/src/ownerWindow.js#L5) from the original `dom-helpers`.

Reply:
@NMinhNguyen Interesting, we should probably use the ownerWindow ownerDocument helpers for all our global event listeners to better support multi-window like use cases. The `parentView` logic originally comes from https://github.com/react-bootstrap/dom-helpers/commit/43fcc2480a22893ff52b0a33baee70d69c0e27f4#diff-991ea3fb4ae30d4ee9eda4d4f6ea9595. I would propose we remove it. The fallback argument is also never used, we can use the same approach as in the ownerDocument helper.

Reply:
Yeah, except it says `parentWindow` in `dom-helpers` 😅. Regardless, I don't think either of those properties exist: `parentView`, `parentWindow`. At least I couldn't find them when googling.

Reply:
`parentWindow` seems to only be suported by IE and opera. Probably super old and just c&p (see over-polyfilling)

Reply:
Ah you’re right. We could change `parentView` -> `parentWindow` or remove it altogether I guess.

Reply:
dom-helpers support older browsers than we do. If we can remove it, it's even better.

Reply:
@NMinhNguyen Do you want to hit this with a pull request? 💥

Reply:
Yes, will do. I can wait until you merge this PR and then change `window` -> `ownerWindow` as well.